starter_code/src/movies.js

```diff
@@ -2,18 +2,77 @@
 
 // Iteration 1: All rates average - Get the average of all rates with 2 decimals 
 
+function ratesAverage(arr){ 
+let sum = arr.reduce(function(acc, value){
+    return acc + parseFloat(value.rate);
+}, 0); return sum / arr.length;
+};
  
 // Iteration 2: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(arr) {
+    let dramaMovies = arr.filter(function(movie){
+        if(movie.genre.indexOf("Drama") !== -1) {
+            return true;
+        }
+    } ); if(dramaMovies.length === 0) {
+        return 0};
+    let sum = dramaMovies.reduce(function(acc, value){
+        return acc + parseFloat(value.rate);
+    }, 0 );
+    return sum / dramaMovies.length;
+};
+
+
 
 // Iteration 3: Ordering by duration - Order by time duration, ascending (in growing order)
+ function orderByYear(arr) {
+   let sortedMovies = arr.sort(function(a, b){
+         if(a.year !== b.year) {
+             return a.year - b.year; 
+        } return a.title.localeCompare(b.title);
+         }); return sortedMovies;
+ };
 
 
 // Iteration 4: Steven Spielberg. The best? - How many movies did STEVEN SPIELBERG direct
 
+function howManyMovies(arr) {
+    let stevenMovies = arr.filter(function(movie){
+        if(movie.genre.indexOf("Drama") !== -1 && movie.director === "Steven Spielberg") {
```

Comment:
you could use the includes method here which will return a boolean, so you then don't have to write !== -1